In [2]:
import tensorflow
import pandas as pd
import numpy as np
import wget
import shutil

In [16]:
#using command line to download a saved checkpoint
!wget http://download.tensorflow.org/models/object_detection/ssd_mobilenet_v2_coco_2018_03_29.tar.gz -P ./models/research/object_detection/

--2018-07-20 20:30:40--  http://download.tensorflow.org/models/object_detection/ssd_mobilenet_v2_coco_2018_03_29.tar.gz
Resolving download.tensorflow.org (download.tensorflow.org)... 172.217.0.48, 2607:f8b0:4005:802::2010
Connecting to download.tensorflow.org (download.tensorflow.org)|172.217.0.48|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 187925923 (179M) [application/x-tar]
Saving to: './models/research/object_detection/ssd_mobilenet_v2_coco_2018_03_29.tar.gz'

ssd_mobilenet_v2_co 100%[===================>] 179.22M  83.7MB/s    in 2.1s    

2018-07-20 20:30:43 (83.7 MB/s) - './models/research/object_detection/ssd_mobilenet_v2_coco_2018_03_29.tar.gz' saved [187925923/187925923]



In [17]:
#using command line to unzip .tar file
!tar -xzvf ./models/research/object_detection/*.tar.gz -C./models/research/object_detection/

ssd_mobilenet_v2_coco_2018_03_29/checkpoint
ssd_mobilenet_v2_coco_2018_03_29/model.ckpt.meta
ssd_mobilenet_v2_coco_2018_03_29/pipeline.config
ssd_mobilenet_v2_coco_2018_03_29/saved_model/saved_model.pb
ssd_mobilenet_v2_coco_2018_03_29/frozen_inference_graph.pb
ssd_mobilenet_v2_coco_2018_03_29/saved_model/
ssd_mobilenet_v2_coco_2018_03_29/saved_model/variables/
ssd_mobilenet_v2_coco_2018_03_29/model.ckpt.index
ssd_mobilenet_v2_coco_2018_03_29/
ssd_mobilenet_v2_coco_2018_03_29/model.ckpt.data-00000-of-00001


In [19]:
!rm ./models/research/object_detection/ssd_mobilenet_v2_coco_2018_03_29.tar.gz

In [21]:
data_folder = "./models/research/object_detection/data_1/"

In [22]:
shutil.move("./models/research/object_detection/samples/configs/ssd_mobilenet_v2_coco.config",data_folder)

'./models/research/object_detection/data_1/ssd_mobilenet_v2_coco.config'

In [ ]:
item {
  id: 1
  name: 'huracan'
}
item {
  id: 2
  name: 'gallardo'
}
item {
  id: 3
  name: 'aventador'
}
item {
  id: 4
  name: 'murcielago'
}
item {
    id: 5
    name: 'urus'
}
item {
    id: 6
    name: 'not a lambo'
}

## We need to create a .pbtxt file with our new classes, easiest is to edit the existing .pbtxt file in *object_detection/data*

# Finally, we need to edit the config file that we just moved

* on line 9 change num_classes from 90 to however many you have (I have 4 right now)
* on line 141 I changed batch size from 24 to 12
* on line 158 change "PATH_TO_BE_CONFIGURED" to the path and then the name of your model e.g."ssd_mobilenet_v1_coco_2018_01_28/model.ckpt"
* on line 177 change "PATH_TO_BE_CONFIGURED" to the path and then the name of your train.record e.g "data/train.record"
* on line 179 change "PATH_TO_BE_CONFIGURED" to the path and then the name of your pbtxt file e.g "data/lambo_detection.pbtxt"
* on line 190 change "PATH_TO_BE_CONFIGURED" to the path and then the name of your train.record e.g "data/test.record"
* on line 192 change "PATH_TO_BE_CONFIGURED" to the path and then the name of your pbtxt file e.g "data/lambo_detection.pbtxt"

# We should be good to go!

pip3 install --ignore-installed --upgrade https://github.com/lakshayg/tensorflow-build/releases/download/tf1.9.0-ubuntu16.04-py27-py35/tensorflow-1.9.0-cp35-cp35m-linux_x86_64.whl

# Copy and paste these into your terminal once you navigate to research

protoc object_detection/protos/*.proto --python_out=.

export PYTHONPATH=$PYTHONPATH:`pwd`:`pwd`/slim

cd ./object_detection/

python3 train.py --logtostderr --train_dir=data_1/ --pipeline_config_path=data_1/ssd_mobilenet_v2_coco.config


* New Terminal

python3 eval.py --logtostderr --pipeline_config_path=data_1/ssd_mobilenet_v2_coco.config --checkpoint_dir=data_1/ --eval_dir=eval_1/

* New Terminal

tensorboard --logdir=/data_1/, /eval_1/

ssh -L 16006:127.0.0.1:6006 science@206.189.219.24

http://localhost:16006/

# After the model is finished, copy and paste the following into your terminal from the object_detection directory. Don't forget to run the protoc and export commands from above

python3 export_inference_graph.py --input_type image_tensor --pipeline_config_path data_1/ssd_mobilenet_v2_coco.config --trained_checkpoint_prefix data_1/model.ckpt-28737 --output_directory object_detection_graph_28737